sdk/resourcemanager/Azure.ResourceManager/src/Custom/Resources/Models/SubResource.cs

```diff
@@ -8,14 +8,14 @@ namespace Azure.ResourceManager.Resources.Models
     /// <summary>
     /// A class representing a sub-resource that contains only the read-only ID.
     /// </summary>
-    [ReferenceType]
-    public abstract partial class SubResource
+    [PropertyReferenceType]
+    public partial class SubResource
     {
         /// <summary>
         /// Initializes an empty instance of <see cref="SubResource"/> for mocking.
         /// </summary>
         [InitializationConstructor]
-        protected SubResource()
+        public SubResource()
```

Comment:
Should we change the other constructor to internal? OR protected internal like WritableSubResource

Reply:
Updated to `internal` to align with the generated common types, also updated `WritableSubResource`.